core/src/io/anuke/mindustry/net/Packets.java

```diff
@@ -140,51 +140,36 @@ public void read(ByteBuffer buffer) {
             timestamp = buffer.getLong();
         }
     }
-
-    public static class BlockLogSyncPacket implements Packet {
-		public IntMap<ArrayList<EditLog>> editlogs;
 	
+	public static class BlockLogRequestPacket implements Packet {
+		public int x;
+		public int y;
+		public Array<EditLog> editlogs;
+		
 		@Override
 		public void write(ByteBuffer buffer) {
-			int size = editlogs.size;
-			buffer.putInt(size);
-			for(IntMap.Entry<ArrayList<EditLog>> editlog :editlogs.entries()) {
-				
-				String key = String.valueOf(editlog.key);
-				buffer.put((byte) key.getBytes().length);
-				buffer.put(key.getBytes());
+			buffer.putInt(x);
+			buffer.putInt(y);
+			buffer.putInt(editlogs.size);
+			for(EditLog value : editlogs) {
 				
-				ArrayList<EditLog> values = editlog.value;
-				buffer.putInt(values.size());
-				for(EditLog value : values) {
-				 
-					String rawValue = value.toString();
-					buffer.put((byte) rawValue.getBytes().length);
-					buffer.put(rawValue.getBytes());
-				}
+				String rawValue = value.toString();
```

Comment:
You're still writing the edit log as a string, which is space-inefficient. You should write two ints for player ID and block ID and two bytes for rotation and action ordinal instead.